application/services/stripe_client.py

```diff
@@ -33,14 +44,35 @@ def get_created_after(no_of_days_ago):
     return created_after
 
 class StripeClass(object):
+
     def __init__(self, api_key):
         stripe.api_key = api_key
+        self.LOGGER = logger.getLogger(__name__)
+
+    def f(self, func):
+        def stripe_call(*args, **kwargs):
+            if current_user:
```

Comment:
functionality here is not to check for authentication but to get the current user id, if the user is not logged in it will be un-set.
The main reason for this check is so that the code can work outside of flask application context also. look at stripe_wrapper.py